taskvine/src/bindings/python3/ndcctools/taskvine/task.py

```diff
@@ -1162,12 +1172,22 @@ class LibraryTask(Task):
     ##
     # Create a new LibraryTask task specification.
     #
-    # @param self       Reference to the current remote task object.
-    # @param fn         The command for this LibraryTask to run
-    # @param name       The name of this Library.
-    def __init__(self, fn, name):
+    # @param self               Reference to the current remote task object.
+    # @param fn                 The command for this LibraryTask to run
+    # @param library_name       The name of this Library.
+    # @param function_list      A list of functions to be added to the library (used for create_library_from_functions).
+    # @param library_code_path  The path to the library code.
+    def __init__(self, fn, library_name, function_list=None, library_code_path=None):
         Task.__init__(self, fn)
+        self.library_name = library_name
+        self.function_list = function_list
+        self.library_code_path = library_code_path
         self._manager_will_free = True
-        self.provides_library(name)
+        self.provides_library(self.library_name)
+
+    def get_function_names(self):
+        if self.function_list:
+            return [f.__name__ for f in self.function_list]
```

Comment:
I'm not sure this works in general.  A library is not required to declare its functions up-front. The function names are only known when using `poncho_serverize`, but this is not the most general case.

This pr should only check if the library with a name exists.  If the function does not exist at the library should be an error reported by the worker (in another pr).